pkg/config/validation/validation.go

```diff
@@ -1514,21 +1523,55 @@ func ValidateAuthorizationPolicy(_, _ string, msg proto.Message) error {
 }
 
 // ValidateRequestAuthentication checks that request authentication spec is well-formed.
-func ValidateRequestAuthentication(_, _ string, msg proto.Message) error {
+func ValidateRequestAuthentication(name, namespace string, msg proto.Message) error {
 	in, ok := msg.(*security_beta.RequestAuthentication)
 	if !ok {
 		return errors.New("cannot cast to RequestAuthentication")
 	}
-	// TODO(diemtvu) add more details validation.
+
 	var errs error
+	errs = appendErrors(errs, validateWorkloadSelector(in.Selector))
+
 	for _, rule := range in.JwtRules {
-		if len(rule.Issuer) == 0 {
-			errs = appendErrors(errs, fmt.Errorf("issuer must be set"))
-		}
+		errs = appendErrors(errs, validateJwtRule(rule))
 	}
 	return errs
 }
 
+func validateJwtRule(rule *security_beta.JWT) (errs error) {
+	if rule == nil {
+		return nil
+	}
+	if rule.Issuer == "" {
+		errs = multierror.Append(errs, errors.New("issuer must be set"))
+	}
+	for _, audience := range rule.Audiences {
+		if audience == "" {
+			errs = multierror.Append(errs, errors.New("audience must be non-empty string"))
+		}
+	}
+
+	if len(rule.JwksUri) != 0 {
+		// TODO: do more extensive check (e.g try to fetch JwksUri)
```

Comment:
for this todo, i don't think we should to actual runtime dynamic jwtksuri. validation webhook might have different networking firewall setup and validation results are subject to transient errors.

validation imo, should be static and deterministica based on the config.

Reply:
+1 to this, we shouldn't call out to the network here.

Reply:
Remove todo.

Reply:
Validation needs to be static, otherwise there are all sorts of issues